src/xccl/ProcessGroupXCCL.cpp

```diff
@@ -585,8 +587,25 @@ c10::intrusive_ptr<Work> ProcessGroupXCCL::collective(
     coalescedAsync_ = asyncOp;
   }
 
-  auto stream = xcclStreamsMap_.at(key).first;
-  auto cclstream = xcclStreamsMap_.at(key).second;
+  auto StreamKey = asyncOp ? key
+                           : key + "_" +
+          std::to_string(at::xpu::getCurrentXPUStream(device.index()).id());
+  auto stream = asyncOp ? xcclStreamsMap_.at(StreamKey).first
+                        : at::xpu::getCurrentXPUStream(device.index());
+
+  std::unique_ptr<ccl::stream> cclstream;
+  try {
+    cclstream =
+        std::make_unique<ccl::stream>(xcclStreamsMap_.at(StreamKey).second);
+  } catch (...) {
+    LOG(WARNING) << "Current stream id changed, create new ccl stream";
```

Comment:
Why is it to be a warning? I think `info` should be more suitable. Warning usually means something not so safe and may cause unpredictable result.